packages/browser/src/client.ts

```diff
@@ -69,6 +70,35 @@ export class BrowserClient extends BaseClient<BrowserBackend, BrowserOptions> {
     return super._prepareEvent(event, scope, hint);
   }
 
+  /**
+   * @inheritDoc
+   */
+  public captureEvent(event: Event, hint?: EventHint, scope?: Scope): string | undefined {
+    let eventId: string | undefined = hint && hint.event_id;
+    this._processing = true;
+
+    this._processEvent(event, hint, scope)
+      .then(finalEvent => {
+        // We need to check for finalEvent in case beforeSend returned null
+        eventId = finalEvent && finalEvent.event_id;
+        // We do this here to not parse any requests if they are outgoing to Sentry
+        // We log breadcrumbs if the integration has them enabled
+        if (finalEvent) {
+          const integration = this.getIntegration(Breadcrumbs);
+          if (integration) {
+            integration.addSentryBreadcrumb(finalEvent);
+          }
+        }
+        this._processing = false;
+      })
+      .then(null, reason => {
+        logger.error(reason);
+        this._processing = false;
+      });
```

Comment:
```suggestion
      .then(finalEvent => {
        // We need to check for finalEvent in case beforeSend returned null.
        if (!finalEvent) {
          return;
        }
        eventId = finalEvent.event_id;
        // We do this here to not parse any requests if they are outgoing to Sentry
        // We log breadcrumbs if the integration has them enabled
        const integration = this.getIntegration(Breadcrumbs);
        if (integration) {
          integration.addSentryBreadcrumb(finalEvent);
        }
      })
      .catch(reason => logger.error(reason))
      .finally(() => (this._processing = false));
```